nxdrive/gui/api.py

```diff
@@ -443,6 +444,23 @@ def default_server_url_value(self) -> str:
         """Make daily job better for our developers :)"""
         return getenv("NXDRIVE_TEST_NUXEO_URL", "")
 
+    @pyqtSlot(str, result=bool)
+    def free_disk_space_under_limit(self, path: str) -> bool:
+        remaining_space = shutil.disk_usage(Path(path).parent).free
+
+        # 10737418240 = 10GiB
+        return remaining_space < 10737418240
+
+    @pyqtSlot(str, result=str)
+    def get_used_disk_space(self, path: str) -> str:
```

Comment:
The method should return the disk space used by synchronized data, not the total space used on the FS.

Instead of crawling the local sync folder to compute the total size, you could use https://github.com/nuxeo/nuxeo-drive/blob/c718a428455d167c780756454820af459d23b410/nxdrive/engine/dao/sqlite.py#L1270